internal/discover/hooks.go

```diff
@@ -44,89 +67,128 @@ func (h *Hook) Hooks() ([]Hook, error) {
 	return []Hook{*h}, nil
 }
 
-type HookName string
-
-// disabledHooks allows individual hooks to be disabled.
-type disabledHooks map[HookName]bool
+type Option func(*cdiHookCreator)
 
-const (
-	// HookEnableCudaCompat refers to the hook used to enable CUDA Forward Compatibility.
-	// This was added with v1.17.5 of the NVIDIA Container Toolkit.
-	HookEnableCudaCompat = HookName("enable-cuda-compat")
-	// directory path to be mounted into a container.
-	HookCreateSymlinks = HookName("create-symlinks")
-	// HookUpdateLDCache refers to the hook used to  Update the dynamic linker
-	// cache inside the directory path to be mounted into a container.
-	HookUpdateLDCache = HookName("update-ldcache")
-)
+type cdiHookCreator struct {
+	nvidiaCDIHookPath string
+	disabledHooks     map[HookName]bool
 
-// AllHooks maintains a future-proof list of all defined hooks.
-var AllHooks = []HookName{
-	HookEnableCudaCompat,
-	HookCreateSymlinks,
-	HookUpdateLDCache,
+	fixedArgs []string
```

Comment:
Instead of checking the path on every `Create` invocation, we check it once at construction.